arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,228 @@
+                            self.selectedFeature = nil
+                            // Present the alert.
+                            self.presentAlert(title: "Make a different selection", message: "Select straight (single segment) polylines only.")
+                            return
+                        }
+                    }
+                    // If the selected feature is a valid polyline, set the value to true.
+                    self.selectedFeatureIsPolyline = true
+                default:
+                    return
+                }
+                // Select the feature.
+                featureLayer.select(selectedFeature)
+                // Set the feature globally.
+                self.selectedFeature = selectedFeature
+            }
+        }
+    }
+    
+    // Create an alert dialog with edit texts to allow editing of the given feature's 'AD_ADDRESS' and 'ST_STR_NAM' attributes.
+    func showEditableAttributes(selectedFeature: AGSFeature) {
+        // Create an alert controller and customize the title and message.
+        let alert = UIAlertController(title: "Edit feature attributes", message: "Edit the 'AD_ADDRESS' and 'ST_STR_NAM' attributes.", preferredStyle: .alert)
+        // Add text fields to prompt user input.
+        alert.addTextField()
+        alert.addTextField()
+        // Populate the text fields with the current values.
+        alert.textFields?[0].text = (selectedFeature.attributes["AD_ADDRESS"] as! NSNumber).stringValue
+        alert.textFields?[1].text = selectedFeature.attributes["ST_STR_NAM"] as? String
+        // Prompt the appropriate keyboard types.
+        alert.textFields?[0].keyboardType = .numberPad
+        alert.textFields?[1].keyboardType = .default
+        // Add a "Cancel" option and clear the selection if cancel is selected.
+        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel) { _ in
+            self.clearSelection()
+            self.selectedFeature = nil
+        })
+        // Add a "Done" option to complete the editing process and close the alert.
+        alert.addAction(UIAlertAction(title: "Done", style: .default) { _ in
+            let addressTextField = alert.textFields?[0]
+            let streetTextField = alert.textFields?[1]
+            // Set the new values for the attributes and update the selected feature.
+            selectedFeature.attributes["AD_ADDRESS"] = Int((addressTextField?.text)!)
+            selectedFeature.attributes["ST_STR_NAM"] = streetTextField?.text
+            selectedFeature.featureTable?.update(selectedFeature)
+        })
+        // Present the alert.
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    // Move the currently selected point feature to the given map point, by updating the selected feature's geometry and feature table.
+    func movePoint(mapPoint: AGSPoint) {
+        guard let selectedFeature = self.selectedFeature else { return }
```

Comment:
`self` isn't needed here.
```suggestion
        guard let selectedFeature = selectedFeature else { return }
```